File: api/src/main/java/org/pmiops/workbench/db/dao/DataSetDao.java
package org.pmiops.workbench.db.dao;

import java.util.List;
import java.util.Optional;
import org.pmiops.workbench.db.model.DbDataset;
import org.springframework.data.repository.CrudRepository;

public interface DataSetDao extends CrudRepository<DbDataset, Long> {
  List<DbDataset> findByWorkspaceIdAndInvalid(long workspaceId, boolean invalid);

  List<DbDataset> findDataSetsByCohortIdsAndWorkspaceIdAndInvalid(
      long cohortId, long workspaceId, boolean dirty);

  List<DbDataset> findDbDataSetsByCohortIdsAndWorkspaceId(long cohortId, long workspaceId);

  List<DbDataset> findDbDatasetsByConceptSetIdsAndWorkspaceId(long conceptId, long workspaceId);

  List<DbDataset> findDataSetsByConceptSetIdsAndWorkspaceIdAndInvalid(
      long conceptId, long workspaceId, boolean dirty);

  List<DbDataset> findByWorkspaceId(long workspaceId);

  Optional<DbDataset> findByDataSetIdAndWorkspaceId(long dataSetId, long workspaceId);

  int countByWorkspaceId(long workspaceId);
}

File: api/src/main/java/org/pmiops/workbench/db/dao/DataSetDao.kt
package org.pmiops.workbench.db.dao

import org.pmiops.workbench.db.model.DataSet
import org.springframework.data.repository.CrudRepository

interface DataSetDao : CrudRepository<DataSet, Long> {
    fun findByWorkspaceIdAndInvalid(workspaceId: Long, invalid: Boolean): List<DataSet>

    fun findDataSetsByCohortIds(cohortId: Long): List<DataSet>

    fun findDataSetsByConceptSetIds(conceptId: Long): List<DataSet>

    fun findByWorkspaceId(workspaceId: Long): List<DataSet>
}
